pkg/skaffold/build/buildpacks/lifecycle.go

```diff
@@ -231,3 +238,23 @@ func resolveDependencyImages(artifact *latestV1.BuildpackArtifact, r ArtifactRes
 
 	return builderImage, runImage, pullPolicy
 }
+
+func containerConfig(artifact *latestV1.BuildpackArtifact) (pack.ContainerConfig, error) {
+	var vols []string
+	if artifact.Volumes != nil {
+		for _, v := range *artifact.Volumes {
+			if v.Host == "" || v.Target == "" {
+				// in case these slip by the JSON schema
```

Comment:
Should we do some sanity check to make sure, the env variable is set to the target location? Could be done as a follow up PR. 

But following checks
1) Warning: You have volume mount set, but no env variable to use the volume?
2) Error: If there is a mismatch in the target specified and the env variable set.
We can define a allowlist of variables you listed in the PR description.

Reply:
I think this risks a lot of false positives.  And there may be situations where just the mount itself is enough to trigger behavioural changes.